core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -587,7 +587,7 @@ class LogManager(logDirs: Seq[File],
     } {
       try {
         checkpoint.write(partitionToLog.mapValues(_.recoveryPoint))
-        allLogs.foreach(_.deleteSnapshotsAfterRecoveryPointCheckpoint())
+        partitionToLog.values.foreach(_.deleteSnapshotsAfterRecoveryPointCheckpoint())
```

Comment:
This is a good fix and we should keep it. However, I think @junrao was suggesting something even better. We call `checkpointLogRecoveryOffsetsInDir` from a number of places where we just want to do it for a single partition. In such cases, we can do `deleteSnapshotsAfterRecoveryPointCheckpoint` for that partition only.